java/ql/lib/semmle/code/java/security/RequestForgery.qll

```diff
@@ -83,3 +84,79 @@ private class HostnameSanitizingPrefix extends InterestingPrefix {
 private class HostnameSantizer extends RequestForgerySanitizer {
   HostnameSantizer() { this.asExpr() = any(HostnameSanitizingPrefix hsp).getAnAppendedExpression() }
 }
+
+/**
+ * An argument to a call to `List.contains()` that is a sanitizer for URL redirects.
+ */
+private predicate isContainsUrlSanitizer(Guard guard, Expr e, boolean branch) {
+  guard =
+    any(MethodCall method |
+      method.getMethod().getName() = "contains" and
+      e = method.getArgument(0) and
+      branch = true
+    )
+}
```

Comment:
The QLDoc says `List.contains`, but the code only checks that `method` has the name `contains`. I'd add the restriction for the type as well, but then the QLDoc below (`ContainsUrlSanitizer`) would need to change.

```suggestion
/**
 * An argument to a call to `List.contains()` that is a sanitizer for URL redirects.
 */
private predicate isContainsUrlSanitizer(Guard guard, Expr e, boolean branch) {
  guard =
    any(MethodCall call |
      call.getMethod().hasQualifiedName("java.util", "List", "contains") and
      e = call.getArgument(0) and
      branch = true
    )
}
```

Reply:
I did the same thing in C#. And there we ended up keeping the sanitizer broad, to support any `.contains()` method, and not just the one on `java.util.List`.  
I'll add a comment if you agree that we can keep the sanitizer broad (that's what we did for C#).

Reply:
As long as the QLDoc is consistent, I'm fine with that (although being less restrictive with sanitizers means more potential for FNs in exchange for less FPs, which goes a bit against our current guidelines).